commonmark/src/main/java/org/commonmark/parser/Parser.java

```diff
@@ -48,6 +50,16 @@ public Node parse(String input) {
         }
         return document;
     }
+    
+    public Node parse(Reader input) throws IOException {
+        InlineParserImpl inlineParser = new InlineParserImpl(specialCharacters, delimiterCharacters, delimiterProcessors);
+        DocumentParser documentParser = new DocumentParser(blockParserFactories, inlineParser);
+        Node document = documentParser.parse(input);
+        for (PostProcessor postProcessor : postProcessors) {
+            document = postProcessor.process(document);
+        }
+        return document;
```

Comment:
Could you extract the above 4 lines into a private method and also use it from the other `parse` method? Otherwise we're duplicating this.